mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,225 @@
+import reactor.core.publisher.Mono;
+
+@ExtendWith(TemporaryFolderExtension.class)
+class DefaultMailboxBackupTest implements MailboxMessageFixture {
+
+    private static final String USER = "user";
+    private static final String OTHER_USER = "otherUser";
+
+    private static final User USER1 = User.fromUsername(USER);
+    private static final User USER2 = User.fromUsername(OTHER_USER);
+
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX1 = MailboxPath.forUser(USER, MAILBOX_1_NAME);
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX2 = MailboxPath.forUser(USER, MAILBOX_2_NAME);
+    private static final MailboxPath MAILBOX_PATH_OTHER_USER_MAILBOX1 = MailboxPath.forUser(OTHER_USER, MAILBOX_OTHER_USER_NAME);
+    private static final HashSet<PreDeletionHook> PRE_DELETION_HOOKS = new HashSet<>();
+
+    private final ArchiveService archiveService = new Zipper();
+    private final MailArchivesLoader archiveLoader = new ZipArchivesLoader();
+
+    private MailboxManager mailboxManager;
+    private BlobStore store;
+    private File destination;
+    private DefaultMailboxBackup backup;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolderExtension.TemporaryFolder temporaryFolder) throws Exception {
+        destination = File.createTempFile("backup-test", ".zip", temporaryFolder.getTempDir());
+        mailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(PRE_DELETION_HOOKS);
+        store = new MemoryBlobStore(new HashBlobId.Factory());
+        backup = new DefaultMailboxBackup(mailboxManager, archiveService, archiveLoader, store);
+    }
+
+    private void readFromStoreAndCopyInFile(BlobId blobId) throws Exception {
+        InputStream content = store.read(blobId);
+        try (OutputStream out = new FileOutputStream(destination)) {
+            IOUtils.copy(content, out);
+        }
+    }
+
+    private void createMailBoxWithMessage(MailboxSession session, MailboxPath mailboxPath, MailboxMessage... messages) throws Exception {
+        MailboxId mailboxId = mailboxManager.createMailbox(mailboxPath, session).get();
+        Arrays.stream(messages).forEach(Throwing.consumer(message ->
+                mailboxManager.getMailbox(mailboxId, session).appendMessage(MessageManager.AppendCommand.from(message.getFullContent()), session)
+            )
+        );
+    }
+
+    private ZipAssert assertThatZipFromIdContainsOnly(Publisher<BlobId> blobIdPublisher, EntryChecks... onlyEntriesMatching) throws Exception {
+        BlobId blobId = Mono.from(blobIdPublisher).block();
+        readFromStoreAndCopyInFile(blobId);
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            return ZipAssert.assertThatZip(zipFile).containsOnlyEntriesMatching(onlyEntriesMatching);
+        }
+    }
+
+    @Test
+    void doBackupWithoutMailboxShouldStoreEmptyBackup() throws Exception {
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res);
```

Comment:
Hidding asserts is bad... Because reading such an assert is not straight forward.

 - 1. the "empty" argument to assert empty content is hidden. It took me a bit of time to realize what is going on
 - 2. I need to read and fully understand assertThatZipFromIdContainsOnly
 - 3. We hide the easy to understand assertJ logic behing a criptic method name...

The alternative is not that long:

```
try (ZipAssert zipAssert = ZipAssert.assertThatZip(blobStore.read(blobId)) {  // we may need an additional helper method there...
    zipAssert.containsOnlyEntriesMatching(onlyEntriesMatching);
}
```

(IMO building a Zip assert from any InputStream is a better way to go!)